pyproject.toml

```diff
@@ -35,7 +31,19 @@ classifiers = [
 dynamic = ["version"]
 
 [project.optional-dependencies]
-inference = ["onnxruntime==1.13.1"]
+all = [
+    "mpi4py>=3.1.4",
+    "netcdf4>=1.6.3",
```

Comment:
Why is `netcdf4` listed as an optional dependency but `h5py` is listed as required? How are we handling the various `xarray` backends? @loliverhennigh is there a way we can either require only `zarr` or should we go all the way and require `netcdf4`, `h5py`, and `zarr`. I guess I am just looking for some clarity here.

Reply:
+1 for centralizing to Zarr for all tensorized data, but what about graph/mesh data? Seems tricky to pick a catch all format.

What is Xarray needed for again in modulus core that the zarr base library cant handle?

Reply:
Only zarr, xarray, fsspec and s3fs are the required deps so we can lean into using these libs for any future datapipes. H5py is not optional